apps/src/templates/manageStudents/MoveStudents.jsx

```diff
@@ -0,0 +1,223 @@
+  static propTypes = {
+    studentData: PropTypes.array.isRequired
+  };
+
+  state = {
+    isDialogOpen: false,
+    selectedIds: []
+  };
+
+  openDialog = () => {
+    this.setState({isDialogOpen: true});
+  };
+
+  closeDialog = () => {
+    this.setState({
+      isDialogOpen: false,
+      selectedIds: []
+    });
+  };
+
+  getStudentIds = () => {
+    return this.props.studentData.map(s => s.id);
+  };
+
+  areAllSelected = () => {
+    return Immutable.Set(this.state.selectedIds).isSuperset(this.getStudentIds());
+  };
+
+  toggleSelectAll = () => {
+    if (this.areAllSelected()) {
+      this.setState({selectedIds: []});
+    } else {
+      this.setState({selectedIds: this.getStudentIds()});
+    }
+  };
+
+  toggleStudentSelected = (studentId) => {
+    let selectedIds = [...this.state.selectedIds];
+
+    if (this.state.selectedIds.includes(studentId)) {
+      const studentIndex = selectedIds.indexOf(studentId);
+      selectedIds.splice(studentIndex, 1);
+    } else {
+      selectedIds.push(studentId);
+    }
+
+    this.setState({selectedIds});
+  };
+
+  selectedStudentHeaderFormatter = () => {
+    return (
+      <input
+        style={styles.checkbox}
+        type="checkbox"
+        checked={this.areAllSelected()}
+        onChange={this.toggleSelectAll}
+      />
+    );
+  };
+
+  selectedStudentFormatter = (_, {rowData}) => {
+    const isChecked = this.state.selectedIds.includes(rowData.id);
+
+    return (
+      <input
+        style={styles.checkbox}
+        type="checkbox"
+        checked={isChecked}
+        onChange={() => this.toggleStudentSelected(rowData.id)}
+      />
+    );
+  };
+
+  getColumns = (sortable) => {
+    return [{
+      property: 'selected',
+        header: {
+          label: '', // TODO: what should this label be?,
+          format: this.selectedStudentHeaderFormatter,
+          props: {
+            style: {
+            ...tableLayoutStyles.headerCell,
+            ...styles.checkboxCell
+          }},
+        },
+        cell: {
+          format: this.selectedStudentFormatter,
+          props: {
+            style: {
+            ...tableLayoutStyles.cell,
+            ...styles.checkboxCell
+          }}
+        }
+      },
+      {
+      property: 'name',
+        header: {
+          label: i18n.name(),
+          props: {
+            style: {
+              ...tableLayoutStyles.headerCell,
+          }},
+          transforms: [sortable],
+        },
+        cell: {
+          props: {
+            style: {
+              ...tableLayoutStyles.cell
+          }}
+        }
+      }];
+    };
+
+  getSortingColumns = () => {
+    return this.state.sortingColumns || {};
+  };
+
+  // The user requested a new sorting column. Adjust the state accordingly.
+  onSort = (selectedColumn) => {
+    this.setState({
+      sortingColumns: sort.byColumn({
+        sortingColumns: this.state.sortingColumns,
+        // Custom sortingOrder removes 'no-sort' from the cycle
```

Comment:
We use this in eleven places now. Let's extract it somewhere!  Maybe into `wrapped_sortable.js` (which should be renamed to camelCase).